custom_components/tuya_local/devices/yym_805SW_aroma_nightlight.yaml

```diff
@@ -1,24 +1,114 @@
-name:  YYM-805SW Aroma diffuser
+# DPS sample: {'updated_at': 1674433099.9169242, '1': True, '2': 'large', '3': '1', '5': True, '6': 'colourful1', '8': '001fff00e8ffff'}
+name: YYM-805SW Aroma diffuser
 products:
-    - id: c0nh3LmEk0NDebrq
+  - id: c0nh3LmEk0NDebrq
 primary_entity:
   entity: fan
   dps:
     - id: 1
       name: switch
       type: boolean
+      mapping:
+        - dps_val: true
+          icon: mdi:scent
+          value: ON
+        - dps_val: false
+          icon: mdi:scent-off
+          value: OFF
     - id: 2
       name: speed
       type: string
+      hidden: true
```

Comment:
Speed should not be hidden, it is a basic part of the fan interface.

Reply:
This is not actually fan so it doesn't have speed But fan device requires speed. Since that just doesn't make a sense that is why I've hide this and created new property for diffuser mode.

Reply:
It does make sense to model it as a fan though.   The fan entity displays in the entities card as a slider, that lets you adjust from off to maximum output, which is quite an intuitive way to control such a device.  Otherwise you are just left with a bundle of switches and select entities which cannot leverage any specialised UI features.